File: build_tmp/src/data/helpContent.js
export const HELP_CONTENT = {
  dashboard: {
    title: "Dashboard",
    overview:
      "Live overview and analytics for your newsagent business. Stat cards show key numbers at the top. The Analytics section shows customers per round, balances by round, publication types, payment status, and top debtors. Auto-refreshes every 60 seconds.",
    how_to: [
      ["View analytics charts", "Scroll below the stat cards to see bar charts and payment status."],
      ["View outstanding balances", "Check the red Outstanding card or the Top Balances table."],
      ["Export from dashboard", "Use Export in the bottom bar for PDF, CSV, or Excel."],
      ["Refresh data", "Click Refresh top-right or press F5."],
    ],
    shortcuts: [
      ["F1", "Open help"],
      ["F5", "Refresh dashboard"],
      ["Ctrl+1", "Go to Dashboard"],
      ["Ctrl+2", "Go to Customers"],
    ],
    faq: [
      ["How often does data refresh?", "Press F5 or click Refresh for immediate update."],
    ],
  },
  customers: {
    title: "Customers",
    overview:
      "Manage customer accounts. Address and phone are encrypted in the database. Balance shown in red when owed.",
    how_to: [
      ["Add customer", "Click New or Ctrl+N, fill Name (required), Save & Close."],
      ["Edit customer", "Select a row, click Edit or Ctrl+E."],
      ["Delete customer", "Select row, Delete key or trash button."],
    ],
    shortcuts: [
      ["F1", "Help"],
      ["Ctrl+N", "New"],
      ["Ctrl+E", "Edit"],
      ["Ctrl+S", "Save form"],
      ["F10", "Quick save snapshot"],
      ["Ctrl+Shift+E", "Export menu"],
      ["Delete", "Delete selected"],
    ],
    faq: [
      ["Is my data secure?", "Address and phone are encrypted at rest with Fernet."],
    ],
  },
  products: {
    title: "Publications / Products",
    overview: "Manage your newspaper and magazine catalog with prices and suppliers.",
    how_to: [
      ["Add publication", "New → name, type, price → Save."],
      ["Update price", "Edit row and change price field."],
    ],
    shortcuts: [
      ["F1", "Help"],
      ["Ctrl+N", "New"],
      ["Ctrl+E", "Edit"],
    ],
    faq: [],
  },
  rounds: {
    title: "Delivery Rounds",
    overview: "Manage delivery routes. Customers can be assigned to a round.",
    how_to: [
      ["Create round", "New → name and type → Save."],
      ["Delete round", "Remove all customers from round first."],
    ],
    shortcuts: [
      ["F1", "Help"],
      ["Ctrl+N", "New"],
    ],
    faq: [
      ["Cannot delete round?", "Reassign or remove customers on that round first."],
    ],
  },
  settings: {
    title: "Settings",
    overview: "Export data, nightly backups at 10 PM, and keyboard shortcuts reference.",
    how_to: [
      ["Manual export", "Ctrl+E or Export button → JSON daily summary."],
      ["Nightly backup", "Runs automatically at 10:00 PM via APScheduler."],
    ],
    shortcuts: [
      ["F10", "Save snapshot"],
      ["Ctrl+E", "Extract / export JSON"],
    ],
    faq: [],
  },
};

File: desktop/modules/help_content.py
MASTER_SHORTCUTS = {

  '── GLOBAL (every screen) ──': {
    'F1':              'Open help window',
    'F2':              'Open help (same as F1)',
    'F5':              'Refresh current view',
    'F9':              'Save / Options menu',
    'F10':             'Main application menu',
    'Escape':          'Close dialog / Cancel',
    'Enter':           'Confirm / Submit',
  },

  '── RECORD OPERATIONS ──': {
    'Ctrl+N':          'New record',
    'Ctrl+O':          'Open / Edit selected',
    'Ctrl+S':          'Save current form',
    'Ctrl+Shift+S':    'Save and add new',
    'Ctrl+R':          'Reset form to original',
    'Ctrl+Delete':     'Delete selected record',
    'Delete':          'Delete selected (lists)',
    'Ctrl+A':          'Select all rows',
    'Ctrl+F':          'Focus search bar',
  },

  '── DATA EXPORT ──': {
    'Ctrl+Shift+X':    'Export (PDF/CSV/Excel)',
    'Ctrl+Shift+P':    'Print current view',
    'Ctrl+Shift+C':    'Copy data to clipboard',
    # NOTE: Ctrl+E REMOVED — caused conflict
  },

  '── APP NAVIGATION ──': {
    'Ctrl+1':          'Dashboard',
    'Ctrl+2':          'Customers',
    'Ctrl+3':          'Rounds & Streets',
    'Ctrl+4':          'Publications',
    'Ctrl+5':          'Billing & Payments',
    'Ctrl+6':          'Suppliers',
    'Ctrl+7':          'Delivery',
    'Ctrl+8':          'Daily Order Sheet',
    'Ctrl+9':          'Analytics',
    'Ctrl+0':          'Settings',
    'Ctrl+W':          'Open Weekly Account',
    'Ctrl+K':          'Open Payment Calendar',
    'Ctrl+L':          'Logout',
  },

  '── WEEKLY ACCOUNT SCREEN ──': {
    'F1 / F2':         'Open help',
    'F3':              'Record payment (this week)',
    'F6':              'Paper & Adjustment screen',
    'F7':              'Amount Received dialog',
    'F8':              'Remove selected paper',
    'F9':              'Save all changes',
    'Ctrl+S':          'Save all changes',
    '0':               'Set quantity to 0 (..)',
    '1 - 9':           'Set quantity in active cell',
    'Tab':             'Move to next day cell',
    'Shift+Tab':       'Move to previous cell',
    'Up / Down':       'Move between paper rows',
    'Left / Right':    'Move between day columns',
    'Ctrl+Shift+P':    'Print weekly account',
    'Ctrl+Shift+C':    'Copy weekly data',
    'Escape':          'Close (save prompt if dirty)',
  },

  '── PAYMENT CALENDAR SCREEN ──': {
    'F1 / F2':         'Open help',
    'F3':              'Mark selected week paid',
    'F6':              'Open Paper & Adjustment',
    'F7':              'Amount Received dialog',
    'F9':              'Options menu',
    'Enter':           'Mark selected week paid',
    'Double-click':    'Mark selected week paid',
    'Arrow keys':      'Navigate calendar cells',
    'PageUp':          'Previous year',
    'PageDown':        'Next year',
    'Ctrl+Shift+P':    'Print calendar',
    'Ctrl+Shift+X':    'Export calendar PDF',
    'Escape':          'Close calendar',
  },

  '── PAPER & ADJUSTMENT SCREEN ──': {
    'F1 / F2':         'Open help',
    'F9':              'Save adjustments',
    'Tab':             'Move to next input field',
    'Shift+Tab':       'Move to previous field',
    'Escape':          'Close (confirm if values)',
  },

  '── AMOUNT RECEIVED DIALOG ──': {
    'F9':              'Confirm and save payment',
    'Enter':           'Confirm and save payment',
    'Tab':             'Next field',
    'Escape':          'Cancel — no changes made',
  },

  '── RECORD VOUCHERS DIALOG ──': {
    'F9':              'Save voucher',
    'Enter':           'Save voucher',
    'Tab':             'Next field',
    'Escape':          'Cancel — no changes made',
  },

  '── TABLE NAVIGATION ──': {
    'Up / Down':       'Navigate rows',
    'Enter':           'Open detail view',
    'Tab':             'Next field/column',
    'Shift+Tab':       'Previous field/column',
    'Ctrl+F':          'Focus search field',
  },
}

HELP_CONTENT = {
  'dashboard': {
    'title': 'Dashboard',
    'overview': 'Live overview of your entire newsagent business. 6 stat cards show key numbers...',
    'how_to': [('Refresh dashboard', 'Press F5 or click Refresh button.')],
    'shortcuts': [('F5', 'Refresh all data')],
    'faq': []
  }
}

def show_help(screen_name):
    pass
